ui/pages/swaps/view-quote/view-quote.js

```diff
@@ -494,50 +501,91 @@ export default function ViewQuote() {
+    current_stx_enabled: currentSmartTransactionsEnabled,
+    stx_user_opt_in: smartTransactionsOptInStatus,
+  };
 
-  useEffect(() => {
-    if (
-      !bestQuoteReviewedEventSent.current &&
-      [
-        sourceTokenSymbol,
-        sourceTokenValue,
-        destinationTokenSymbol,
-        destinationTokenValue,
-        fetchParams,
-        topQuote,
-        numberOfQuotes,
-        feeInFiat,
-      ].every((dep) => dep !== null && dep !== undefined)
-    ) {
-      bestQuoteReviewedEventSent.current = true;
-      trackEvent({
-        event: 'Best Quote Reviewed',
-        category: 'swaps',
-        sensitiveProperties: {
-          token_from: sourceTokenSymbol,
-          token_from_amount: sourceTokenValue,
-          token_to: destinationTokenSymbol,
-          token_to_amount: destinationTokenValue,
-          request_type: fetchParams?.balanceError,
-          slippage: fetchParams?.slippage,
-          custom_slippage: fetchParams?.slippage !== 2,
-          response_time: fetchParams?.responseTime,
-          best_quote_source: topQuote?.aggregator,
-          available_quotes: numberOfQuotes,
-          is_hardware_wallet: hardwareWalletUsed,
-          hardware_wallet_type: hardwareWalletType,
-          stx_enabled: smartTransactionsEnabled,
-          current_stx_enabled: currentSmartTransactionsEnabled,
-          stx_user_opt_in: smartTransactionsOptInStatus,
-          network_fees: feeInFiat,
-        },
-      });
-    }
+  const trackAllAvailableQuotesOpened = () => {
+    trackEvent({
+      event: 'All Available Quotes Opened',
+      category: 'swaps',
+      sensitiveProperties: {
+        ...eventObjectBase,
+        other_quote_selected: usedQuote?.aggregator !== topQuote?.aggregator,
+        other_quote_selected_source:
+          usedQuote?.aggregator === topQuote?.aggregator
+            ? null
+            : usedQuote?.aggregator,
+      },
+    });
+  };
+  const trackQuoteDetailsOpened = () => {
+    trackEvent({
+      event: 'Quote Details Opened',
+      category: 'swaps',
+      sensitiveProperties: {
+        ...eventObjectBase,
+        other_quote_selected: usedQuote?.aggregator !== topQuote?.aggregator,
+        other_quote_selected_source:
+          usedQuote?.aggregator === topQuote?.aggregator
+            ? null
+            : usedQuote?.aggregator,
+      },
+    });
+  };
+  const trackEditSpendLimitOpened = () => {
+    trackEvent({
+      event: 'Edit Spend Limit Opened',
+      category: 'swaps',
+      sensitiveProperties: {
+        ...eventObjectBase,
+        custom_spend_limit_set: originalApproveAmount === approveAmount,
+        custom_spend_limit_amount:
+          originalApproveAmount === approveAmount ? null : approveAmount,
+      },
+    });
+  };
+  const trackBestQuoteReviewedEvent = useCallback(() => {
+    trackEvent({
+      event: 'Best Quote Reviewed',
+      category: 'swaps',
+      sensitiveProperties: {
+        token_from: sourceTokenSymbol,
```

Comment:
looks like we can use `...eventObjectBase` here as well

Reply:
@digiwand Done [here](https://github.com/MetaMask/metamask-extension/pull/13955/commits/1951d4ccd8cf61ef1ca9b36037e7ed9fceb744e3)